app/src/engineering/kotlin/com/hedvig/app/DevelopmentScreenAdapter.kt

```diff
@@ -132,15 +134,19 @@ class DevelopmentScreenAdapter(
 
     class Footer(
       parent: ViewGroup,
-      private val authenticationTokenService: AuthenticationTokenService,
+      private val authTokenService: AuthTokenService,
     ) : ViewHolder(parent.inflate(R.layout.development_footer)) {
       private val binding by viewBinding(DevelopmentFooterBinding::bind)
       override fun bind(data: DevelopmentScreenItem) {
         with(binding) {
-          token.setText(authenticationTokenService.authenticationToken)
+          token.setText(authTokenService.getToken()?.token ?: "Token not found")
           saveToken.setHapticClickListener {
-            authenticationTokenService.authenticationToken = saveToken.text.toString()
-            saveToken.context.makeToast("Token saved")
+            authTokenService.updateTokens(
+              AccessToken(token.text.toString(), 600),
+              // todo maybe the dev screen shouldn't be doing this in the first place.
+              RefreshToken(token.text.toString(), 600),
+            )
+            token.context.makeToast("Token saved")
```

Comment:
I haven't used this feature really myself, do you? I usually use the impersonation instead if I need to do this. Especially with the new login apis I don't know how often we'd want to edit the token here, especially considering we'd have to edit two of them.

*Maybe* we can consider keeping it here to be copied, but not really to be edited?

Reply:
Not using this feature either. We can keep it for now and maybe take an off-piste to remove it later if we want to :)